src/ok.rs

```diff
@@ -261,3 +262,71 @@ fn dev_sev_rw(file: &mut fs::OpenOptions) -> TestResult {
 
     TestResult { success, msg }
 }
+
+fn has_kvm_support() -> TestResult {
+    let mut success = true;
+    let path = "/dev/kvm";
+
+    let m = match File::open(path) {
+        Ok(kvm) => {
+            let api_version = unsafe { libc::ioctl(kvm.as_raw_fd(), 0xAE00, 0) };
+            if api_version < 0 {
+                success = false;
+                "KVM probing error: accessing KVM device node failed".to_string()
+            } else {
+                format!("KVM API version: {}", api_version)
+            }
+        }
+        Err(e) => {
+            success = false;
+            format!(
+                "KVM probing error while reading {}: ({})",
+                path,
+                e.to_string().red()
+            )
+        }
+    };
+
+    TestResult {
+        success,
+        msg: Some(m),
+    }
+}
+
+fn sev_enabled_in_kvm() -> TestResult {
+    let mut success = true;
+    let path_loc = "/sys/module/kvm_amd/parameters/sev";
+    let path = std::path::Path::new(path_loc);
+
+    let msg = if path.exists() {
+        match std::fs::read_to_string(path_loc) {
+            Ok(result) => {
+                if result.trim() == "1" {
+                    format!("SEV enabled in KVM: able to read {}", path_loc)
```

Comment:
The test isn't exclusively checking if we can read from it; it's checking to see if it's enabled "1" or disabled "0".